runtime/vm/CRIUHelpers.cpp

```diff
@@ -1187,6 +1190,63 @@ toggleSuspendOnJavaThreads(J9VMThread *currentThread, U_8 suspendResumeFlag)
 	}
 }
 
+/**
+ * Get the package/class.method array specified via a system property
+ * -Dopenj9.internal.criu.notcheckpointsafe=package1/class1.method1,package2/class2.method2
+ *
+ * Caller must free the returning structure.
+ *
+ * @param[in] vmThread vmThread token
+ * @param[in/out] classMethodNumber the number of package/class.method specified
+ */
+static char**
+getNotCheckpointSafeClassMethodArray(J9VMThread *vmThread, UDATA *classMethodNumber)
+{
+	J9JavaVM *vm = vmThread->javaVM;
+	char **notcheckpointsafeArray = NULL;
+	J9VMSystemProperty *sysPropNotcheckpointsafe = NULL;
+	UDATA rc = getSystemProperty(vm, SYS_PROP_OPENJ9_NOT_CHECKPOINT_SAFE, &sysPropNotcheckpointsafe);
+
+	if (J9SYSPROP_ERROR_NONE == rc) {
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		UDATA classMethodCounter = 0;
+		char *notcheckpointsafeValue = sysPropNotcheckpointsafe->value;
+
+		Trc_VM_criu_getNotCheckpointSafeClassMethodArray_propValue(vmThread, notcheckpointsafeValue);
+		for (UDATA index = 0; index < strlen(notcheckpointsafeValue); index++) {
```

Comment:
Could `notcheckpointsafeValue` be `NULL` or empty string here ? `strlen()` will segfault if it is NULL. `classMethodCounter` will be incorrectly incremented to 1 later if it is an empty string.

Reply:
`notcheckpointsafeValue` is returned from `getSystemProperty()` and can't be `NULL`, most of this method callers don't check for `NULL`.
Updated to exit earlier if it is an empty string, also added a testcase with empty property value.